src/boost_histogram/_internal/axis.py

```diff
@@ -37,6 +37,31 @@ def __copy__(self):
         other._ax = copy.copy(self._ax)
         return other
 
+    def __getattr__(self, item):
+        if item == "_ax":
+            return Axis.__dict__[item].__get__(self)
+        elif self._ax.metadata is not None and item in self._ax.metadata:
```

Comment:
Does it make sense to init `self._ax.metadata` as an empty dict, to save this check for every attribute access?

Reply:
I was trying to be backward compatible, but I think that's actually handled by setstate/getstate properly, so probably yes.

Reply:
I think you can change this without breaking backward compatibility, but perhaps a few unit tests could be added that check for possible corner cases?

Reply:
This is working well, and much simpler, I just am fixing some of the unit tests that checked a raw C++ class (the NumPy one) along with the wrapped axes, expecting them to have the same API.